app/models/user.rb

```diff
@@ -14,4 +17,32 @@ class User < ApplicationRecord
                        length: {minimum: Settings.validates.password_min_length}
 
   has_secure_password
+
+  # Returns the hash digest of the given string.
+  def self.digest string
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                              BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  # Remembers a user in the database for use in persistent sessions.
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end
+
+  def authenticated? remember_token
+    return false unless remember_token
+
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  # Forgets a user.
+  def forget
+    update_attribute(:remember_digest, nil)
```

Comment:
bỏ () những chỗ ko cần thiết
```suggestion
    update_attribute :remember_digest, nil
```